### DISCUSSION TITLE
Easy way to calculate maximum across multiple variables in a dataset

### DISCUSSION THREAD

**Discussion by gewitterblitz** [pydata/xarray#6321] [Q&A] [answered]
Hi,

I have a spatiotemporal dataset (3D space and 1D time) with multiple variables (same dimensions and units) and I would like to calculate which variable has the highest value in each grid cell at each time step. Here's an example:

<img width="796" alt="Screen Shot 2022-03-02 at 9 54 18 AM" src="https://user-images.githubusercontent.com/13985417/156386632-02625788-d1be-473a-9299-0107699713a1.png">

What's the best way to go about doing such an operation?

**Comment by dcherian** [comment, marked as answer]
[`Dataset.to_array`](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.to_array.html) is very useful for this kind of thing. then call [`idxmax`](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.idxmax.html) on the DataArray.

``` python
In [2]: ds = xr.Dataset(
   ...:     {"a": ("x", [0, 1, 2, 3]), "b": ("x", [-1, 2, 0, 0]), "c": ("x", [0, 1, 4, 5])}
   ...: )

In [3]: ds.to_array("variable").idxmax("variable")
Out[3]:
<xarray.DataArray 'variable' (x: 4)>
array(['a', 'b', 'c', 'c'], dtype='<U1')
Dimensions without coordinates: x
```

**Reply by gewitterblitz** [reply]
Thanks, @dcherian, this was helpful!
